### PR TITLE
AG-11922 a11y: Add text elements and set `role="presentation"` on divs

### PR DESCRIPTION
**Pull request by olegat**
https://ag-grid.atlassian.net/browse/AG-11922

### CODE REVIEW

File: packages/ag-charts-community/src/chart/update/baseLayoutProcessor.ts [hunk 1]
```diff
@@ -5,22 +5,31 @@ import type { BBox } from '../../scene/bbox';
...
+        private readonly proxyInteractionService: ProxyInteractionService,
+        readonly domManager: DOMManager
     ) {
         this.destroyFns.push(
             // eslint-disable-next-line sonarjs/no-duplicate-string
             this.layoutService.addListener('start-layout', (e) => this.positionPadding(e)),
             this.layoutService.addListener('layout-complete', (e) => this.alignCaptions(e)),
             this.layoutService.addListener('start-layout', (e) => this.positionCaptions(e))
         );
+        // TODO tidy up
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/update/baseLayoutProcessor.ts]: ?

File: packages/ag-charts-community/src/chart/update/baseLayoutProcessor.ts [hunk 2]
```diff
@@ -135,5 +144,9 @@ export class BaseLayoutProcessor implements UpdateProcessor {
                 caption.node.x = rect.x + rect.width - bbox.width - titlePadding;
             }
         }
+
+        [title, subtitle, footnote].forEach((c) =>
+            c.updateA11yText(this.proxyInteractionService, this.proxyTextContainer)
+        );
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/update/baseLayoutProcessor.ts]: This feels like an odd place to be doing this, as the A11y update doesn't seem related to layout concerns?

Maybe `Chart.updateDOM()` would fit better? (I'm not sure if that is always invoked at the right time though?)

File: packages/ag-charts-community/src/chart/update/baseLayoutProcessor.ts (hunk 1 above)
Comment (olegat) [reply packages/ag-charts-community/src/chart/update/baseLayoutProcessor.ts]: Well spotted! This was a reminder to integrate this with the `DOMManager` properly. I've done this in https://github.com/ag-grid/ag-charts/commit/f9cfa7e4998a284adcfd5492202f207b3ce6a178

Done.

File: packages/ag-charts-community/src/chart/update/baseLayoutProcessor.ts (hunk 2 above)
Comment (olegat) [reply packages/ag-charts-community/src/chart/update/baseLayoutProcessor.ts]: This update is related to layout concerns. If the CSS bounds of the text aren't set correctly then the screen-reader will indicate the incorrect bounds.

---

Screenshot 2024-07-08 at 15 04 32
_Screenshot on https://github.com/ag-grid/ag-charts/commit/f9cfa7e4998a284adcfd5492202f207b3ce6a178._

---

Screenshot 2024-07-08 at 15 19 32
_Screenshot on https://github.com/ag-grid/ag-charts/commit/17a3e1f2c89512baabd08f36f6d20b9940a93d3b._

**Review by alantreadway — APPROVED** [review]
LGTM
